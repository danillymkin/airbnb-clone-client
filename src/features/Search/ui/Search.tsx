import React, { FunctionComponent } from 'react';
import { BiSearch } from 'react-icons/bi';

export const Search: FunctionComponent = (): JSX.Element => {
  return (
    <div
      className={
        'w-full cursor-pointer rounded-full border py-2 shadow-sm transition hover:shadow-md md:w-auto'
      }
      role={'search'}
    >
      <div className={'flex items-center justify-between'}>
        <div className={'px-6 text-sm font-semibold'}>Anywhere</div>
        <div
          className={
            'hidden flex-1 border-x px-6 text-center text-sm font-semibold sm:block'
          }
        >
          Any Week
        </div>
        <div
          className={'flex items-center gap-3 pl-6 pr-2 text-sm text-gray-600'}
        >
          <div className={'hidden sm:block'}>Add Guests</div>
          <div className={'rounded-full bg-rose-500 p-2 text-white'}>
            <BiSearch role={'img'} />
          </div>
        </div>
      </div>
    </div>
  );
};
