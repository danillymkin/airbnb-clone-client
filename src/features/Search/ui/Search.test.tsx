import { render, screen } from '@testing-library/react';
import { Search } from './Search';

describe('Search', () => {
  it('should render search', () => {
    render(<Search />);
    const search = screen.getByRole('search');
    expect(search).toBeTruthy();
  });

  it('should match snapshot', () => {
    render(<Search />);
    const search = screen.getByRole('search');
    expect(search).toMatchSnapshot();
  });

  it('should have anywhere filter', () => {
    render(<Search />);
    const anywhere = screen.getByText(/anywhere/i);
    expect(anywhere).toBeInTheDocument();
  });

  it('should have any week filter', () => {
    render(<Search />);
    const anyWeek = screen.getByText(/any week/i);
    expect(anyWeek).toBeInTheDocument();
  });

  it('should have add guests filter', () => {
    render(<Search />);
    const addGuests = screen.getByText(/add guests/i);
    expect(addGuests).toBeInTheDocument();
  });

  it('should have search icon', () => {
    render(<Search />);
    const icon = screen.getByRole('img');
    expect(icon).toBeInTheDocument();
  });
});
