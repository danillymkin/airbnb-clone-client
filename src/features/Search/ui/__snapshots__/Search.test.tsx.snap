// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match snapshot 1`] = `
<div
  class="w-full cursor-pointer rounded-full border py-2 shadow-sm transition hover:shadow-md md:w-auto"
  role="search"
>
  <div
    class="flex items-center justify-between"
  >
    <div
      class="px-6 text-sm font-semibold"
    >
      Anywhere
    </div>
    <div
      class="hidden flex-1 border-x px-6 text-center text-sm font-semibold sm:block"
    >
      Any Week
    </div>
    <div
      class="flex items-center gap-3 pl-6 pr-2 text-sm text-gray-600"
    >
      <div
        class="hidden sm:block"
      >
        Add Guests
      </div>
      <div
        class="rounded-full bg-rose-500 p-2 text-white"
      >
        <svg
          fill="currentColor"
          height="1em"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
